### PR TITLE
Remove [NoInterfaceObject]

### LINKED ISSUE #19: Remove [NoInterfaceObject]
Carefully consider with what name to expose these interfaces and then please drop this. We're going to remove it from IDL per https://github.com/heycam/webidl/issues/430 and https://github.com/heycam/webidl/pull/609.

### PR DESCRIPTION
**Pull request by anssiko**
Rename exposed interfaces

Fix #19

### CODE REVIEW

**Comment by anssiko** [comment]
PTAL @reillyeon

File: index.html [hunk 1]
```diff
@@ -382,9 +381,9 @@ <h3><dfn>Geolocation</dfn> interface</h3>
    void clearWatch(long watchId);
  };
 
- callback PositionCallback = void (Position position);
+ callback PositionCallback = void (NavigatorGeolocationPosition position);
 
- callback PositionErrorCallback = void (PositionError positionError);
+ callback PositionErrorCallback = void (PositionErrorPositionError positionError);
```
Comment (reillyeon) [inline index.html]: s/PositionErrorPositionError/NavigatorGeolocationPositionError/

Comment (anssiko) [reply index.html]: Amazing typo 🤣

Now fixed.

**Comment by anssiko** [comment]
@honry can help update https://github.com/web-platform-tests/wpt/tree/master/geolocation-API accordingly? TL;DR: this patch exposes some interfaces so maybe idlharness.js helps.

**Comment by Honry** [comment]
@foolip's [bot ](https://github.com/autofoolip)will take care of every IDL updates. (i.e. the bot will automatically generate a PR with updated idlharness.js tests to reflect to every IDL updates in all specs.)

**Comment by anssiko** [comment]
Props to @autofoolip for taking care of idlharness.js tests automatically.

IIUC the bot uses IDLs extracted by reffy (here https://github.com/web-platform-tests/wpt/blob/master/interfaces/geolocation-API.idl) that extracts them by crawling the TR version of the spec (in this case https://www.w3.org/TR/geolocation-API). What follows is that the IDL tests will be updated automatically only when a new spec version is published on TR.

My interpretation is we're still compliant with the test as you commit policy [we follow in this group](https://www.w3.org/2018/06/devices-sensors-wg-charter.html#deliverables). We just need to publish a TR version after any substantial change to the spec IDL to trigger the IDL test update.

@Honry, please correct me if I'm wrong.

**Comment by Honry** [comment]
All refreshed IDLs used by the bot are maintained in https://github.com/tidoust/reffy-reports/tree/master/w3c/idl 

Looks like it crawls the ED version of the spec, for example, the bot updated the https://github.com/web-platform-tests/wpt/pull/15125 while the IDL change is not published to TR yet.

**Comment by anssiko** [comment]
@Honry, thanks for the clarification. Using IDL from EDs as the source sounds like a better solution. No issues then.

**Comment by marcoscaceres** [comment]
Oh wait, did https://github.com/w3c/geolocation-api/pull/23 undo this? If yes, yay 😃

**Comment by marcoscaceres** [comment]
Anyway, please please please file bugs on Gecko with changes you are making, otherwise we are all going to get out of sync!

**Comment by reillyeon** [comment]
#23 partially undid the rename. https://chromium-review.googlesource.com/c/chromium/src/+/1471230 is the change to implement this in Blink which I never merged due to some CQ flakiness. I should get on that.

It still renames,

```
  Coordinates   -> GeolocationCoordinates
  Position      -> GeolocationPosition
  PositionError -> GeolocationPositionError
```

**Comment by marcoscaceres** [comment]
Ok, that’s not too bad... but if we can avoid the renaming entirely 🥰

**Comment by reillyeon** [comment]
Wouldn't you agree that the names "coordinates" and "position" are likely to be confused with rendering concepts from other parts of the platform?

**Comment by marcoscaceres** [comment]
> Wouldn't you agree that the names "coordinates" and "position" are likely to be confused with rendering concepts from other parts of the platform?

From the engines perspective, no - because they are already implemented specifically for Geolocation.

From an authoring perspective, maybe: They don't have constructors, so they can't be directly used anyway. A new set of coordinates from any new spec could just be named `2DCoordinates` and `3DPosition` or whatever, without us needing to rename this one (i.e., this name is already squatted).

**Comment by reillyeon** [comment]
If you are firmly opposed to renaming I agree that the names are already squatted and future APIs can work around it. From my perspective this is our last shot before renaming becomes impossible so it seems like a good opportunity to pick something less likely to cause confusion.

**Comment by anssiko** [comment]
@marcoscaceres said:

>As @annevk mentioned, it’s probably not worth doing this at this point.

Per my reading of https://github.com/w3c/geolocation-api/issues/19#issue-399256646 @annevk is not opposed to the idea of rename, just wanted the group to "carefully consider with what name to expose these interfaces and then please drop this.", where "drop this" refers to `[NoInterfaceObject]`. I feel the group did carefully consider the names as documented in #19 and #23.

To be clear, the following are the interfaces that dropped `[NoInterfaceObject]` and the renames to address #19:
```
Coordinates -> GeolocationCoordinates
Position -> GeolocationPosition
PositionError -> GeolocationPositionError
```

This seems like a good opportunity for implementers to roll both the (mostly) mechanical rename and the removal of `[NoInterfaceObject]` into the same patch as in https://chromium-review.googlesource.com/c/chromium/src/+/1471230

Yours,
Anssi the co-chair

**Comment by marcoscaceres** [comment]
I think we need a third opinion here from the WebKit folks - otherwise this exercise is rather pointless. I'm willing to make the change in Gecko if they are also willing to make the change in WebKit. It's a bunch of refactoring work, so I'd personally prefer to spend time on other things. However, if others think this preemptive change is good for the platform, I'm ok to invest time to do the refactoring.

**Comment by anssiko** [comment]
Looping in @cdumez for WebKit feedback. Here's the Blink patch of this spec change:
https://chromium-review.googlesource.com/c/chromium/src/+/1471230

**Comment by marcoscaceres** [comment]
While we are here, can kindly request that we add:
https://github.com/w3c/screen-orientation/blob/gh-pages/.github/PULL_REQUEST_TEMPLATE.md

That way, we make sure nothing gets merged into the spec unless we have agreement from at least 2 browser vendors, browser bugs are filed, and we have tests?

**Comment by cdumez** [comment]
This looks reasonable. Where is the WebKit bug so that I can work on this?

**Comment by marcoscaceres** [comment]
Curse you, @cdumez 😋 guess I’ll be refactoring too.

**Comment by anssiko** [comment]
@cdumez, the WebKit bug is at https://webkit.org/b/200885 Thanks!

@marcoscaceres, you want me to open one for Gecko? You'll know Mozillians who should be cc'd in the bug, but I'm happy to create one if so preferred.

**Comment by marcoscaceres** [comment]
That would be great! Please cc me and we will go from there.

**Comment by anssiko** [comment]
@marcoscaceres, the Gecko bug is at https://bugzil.la/1575144 Thanks!

**Comment by marcoscaceres** [comment]
Awesome, thanks!

**Comment by anssiko** [comment]
The WebKit bug https://webkit.org/b/200885 is now fixed. That was fast, thanks @cdumez!

**Comment by marcoscaceres** [comment]
Landed in Gecko https://bugzilla.mozilla.org/show_bug.cgi?id=1575144

**Pull request merged**